miden-lib/src/notes/mod.rs

```diff
@@ -20,6 +20,14 @@ pub enum Script {
         target: AccountId,
         recall_height: u32,
     },
+    ASWAP {
+        faucet_id: AccountId,
+        amount: Felt,
+        tag: Felt,
+        // Should be -
+        // recipient: Digest,
+        recipient: Felt,
+    },
```

Comment:
I think this can be simplified to be just:
```Rust
ASWAP {
    asset: Asset,     // asset to be sent in return
    serial_num: Word, // serial_num for the return note
}
```
Everything else we can build from these + the parameters passed into `create_note()`. Specifically:

1. `tag` should be derived from the sender account ID.
2. `recipient` should be built from `serial_num`, script root of `P2ID` script, and inputs for the `P2ID` script.

On a separate note: I don't love that we need to pass serial numbers explicitly (both here and in `create_note()`). Probably a better way is to define an `Rng` interface which can generate random words. Then we can pass an `Rng` object into `create_note()` and generate all the required serial numbers as needed. I'll create a separate issue for this and we can address this in another PR.

Reply:
> 2. `recipient` should be built from `serial_num`, script root of `P2ID` script, and inputs for the `P2ID` script.

Quick question @bobbinth: We don't explicitly use the P2ID script here. We ensure explicit consumption by the initial sender using `RECIPIENT` rather than the P2ID script, right?

Reply:
Technically, return script does not need to be `P2ID` but I think we can simplify and make it so here (and if someone wants to build a swap with a more complex return script, they can do so manually).